### PR TITLE
PM4 Support.

### PR DESCRIPTION
**Pull request by JaxkDev**
- Main Class Lint adjusted to account for src-namespace-prefix
- PSR-0 Lint not specified when PSR-4 (src-namespace-prefix) is used.

### CODE REVIEW

File: src/poggit/ci/builder/ProjectBuilder.php [hunk 1]
```diff
@@ -570,7 +570,8 @@ protected function lintManifest(RepoZipball $zipball, BuildResult $result, strin
             $status->className = $manifest["main"];
             $result->addStatus($status);
         }
-        if(!$zipball->isFile($mainClassFile = $this->project->path . "src/" . str_replace("\\", "/", $mainClass = $manifest["main"]) . ".php")) {
+        $mainClassFile = $this->project->path . "src/" . str_replace((trim($manifest["src-namespace-prefix"]??"") === "" ?: str_replace("\\", "/", $manifest["src-namespace-prefix"])."/"), "", str_replace("\\", "/", $mainClass = $manifest["main"])) . ".php";
```
Comment (SOF3) [inline src/poggit/ci/builder/ProjectBuilder.php]: Could you split down this line please? Poggit doesn't really have good code style, but we don't have to torture ourselves further with a 250-character-long line.

**Review by SOF3 — CHANGES_REQUESTED** [review]

File: src/poggit/ci/builder/DefaultProjectBuilder.php [hunk 2]
```diff
@@ -170,7 +171,8 @@ protected function build(Phar $phar, RepoZipball $zipball, WebhookProjectModel $
             $localName = $out . substr($file, strlen($in));
             $phar->addFromString($localName, $contents = $reader());
             if(Lang::startsWith($localName, "src/") and Lang::endsWith(strtolower($localName), ".php")) {
-                $this->lintPhpFile($result, $localName, $contents, $localName === $mainClassFile, ($doLint ? $project->manifest["lint"] : null));
+                $this->lintPhpFile($result, $localName, $contents, $localName === $mainClassFile,
+                    $pluginManifest["src-namespace-prefix"]??"", ($doLint ? $project->manifest["lint"] : null));
```
Comment (dktapps) [inline src/poggit/ci/builder/DefaultProjectBuilder.php]: src-namespace-prefix should be ignored if we're not on PM4.

Also, use of src-namespace-prefix should be a lint error on PM3 API plugins.

File: src/poggit/ci/builder/ProjectBuilder.php (hunk 1 above)
Comment (JaxkDev) [reply src/poggit/ci/builder/ProjectBuilder.php]: Better?

File: src/poggit/ci/builder/DefaultProjectBuilder.php (hunk 2 above)
Comment (JaxkDev) [reply src/poggit/ci/builder/DefaultProjectBuilder.php]: API Parsing is not one of poggit's strong suite I'll see.

Defo haven't got time to be messing with new lints for api specific plugins rn (same could be said about OS field < 3.12.0)

**Comment by JaxkDev** [comment]
Virions will take some time I don't have the time today to even look at what happens where, damn biological words don't help.

**Comment by JaxkDev** [comment]
Not the best methods of doing this but I literally am out of time this is the only time i had until at least 11th Dec.

**Pull request merged**